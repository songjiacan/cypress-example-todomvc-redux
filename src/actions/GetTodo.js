import * as types from '../constants/ActionTypes'
import axios from 'axios'
import {APIURL} from '../constants/EndPoint'

// get data from rest service
export const getTodoApi = () => {
  // const apiUrl = 'http://jsonplaceholder.typicode.com/todos';
     return  fetch(APIURL)
     .then(response => response.json())
     .then(data => {
       console.log(data)
       return data
     })
     .catch(err => `${err.message}`);
     
}

export const fetUserSucess = todos => {
  return {
    type: types.LOAD_DB_TODO,
    payload: todos
  }
}



export const addTodoToDB = (body)=> {
  return (dispatch) => {
    axios.post(APIURL, body)
    .catch(err => `${err.message}`);
  }
}

export const deleteTodoToDB = (id)=> {
  axios.delete(APIURL + `/${id}`)
      .then(res => {
        console.log(res);
        console.log(res.data);
      })
}

export const updateTodoToDB = (id, body)=> {
  axios.put(APIURL + `/${id}`, body)
      .then(res => {
        console.log(res);
        console.log(res.data);
      })
}
// axios({
//   method: 'post',
//   url: '/user/12345',
//   data: {
//     firstName: 'Fred',
//     lastName: 'Flintstone'
//   }
// });

export const fetchTodos = ()=> {
  // const apiUrl = 'http://jsonplaceholder.typicode.com/todos';
  return (dispatch) => {
    //  fetch(APIURL)
    //     .then(res => res.json())
    //     .then(json => {
    //       console.log(json)
    //       dispatch(fetUserSucess(json))
    //     });

    axios.get(APIURL)  
    .then( response => {
      const todoBody = response.data
      console.log("data " + todoBody)
      dispatch(fetUserSucess(todoBody))

    })
    .catch(err => `${err.message}`);
  }

}


// fetch(`/api/titles/${titleNo}`)
//         .then(res => res.json())
//         .then(json => {
//           console.log(json)
//           this.setState({
//             data: json,
//           })
//         });

// get getDBTodo action
// export function getDBTodo() {
//   return (dispatch) => {
//     dispatch({
//       type: types.LOAD_DB_TODO,
//       payload: getTodoApi(),
//     });
//   };
// }

export default function getDBTodo() {
  return {
      type: types.LOAD_DB_TODO,
      payload: getTodoApi(),
    }
  }