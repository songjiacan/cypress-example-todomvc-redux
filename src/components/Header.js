import PropTypes from 'prop-types'
import React from 'react'
import TodoTextInput from './TodoTextInput'

const Header = ({maxId, addTodo, fetchTodos,addTodoToDB }) => (
  <header className='header'>
    <h1>todos</h1>
    <TodoTextInput
      getDBTodo = {fetchTodos}
      newTodo
      onSave={text => {
        if (text.length !== 0) {
          // simulate delayed application logic
          // setTimeout(addTodo, 1000, text)
          addTodo(text)
          addTodoToDB({id: maxId, text:text, completed: false})
        }
      }}
      placeholder='What needs to be done?'
    />
  </header>
)

Header.propTypes = {
  addTodo: PropTypes.func.isRequired
}

export default Header
