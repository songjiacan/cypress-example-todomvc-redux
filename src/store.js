import React from 'react'
import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import reducer from './reducers'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import {composeWithDevTools} from 'redux-devtools-extension'

export const store = createStore(
  reducer,
  composeWithDevTools(applyMiddleware(logger, thunk)))
  // window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
export const StoreProvider = ({children}) => (<Provider store={store}>{children}</Provider>)

// expose store during tests
/* istanbul ignore else */
if (window.Cypress) {
  window.store = store
}
