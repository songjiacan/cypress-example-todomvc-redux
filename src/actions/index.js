import * as types from '../constants/ActionTypes'
// import {getTodoApi} from './GetTodo'
import {fetchTodos} from './GetTodo'



export const addTodo = text => ({ type: types.ADD_TODO, text })
export const deleteTodo = id => ({ type: types.DELETE_TODO, id })
export const editTodo = (id, text) => ({ type: types.EDIT_TODO, id, text })
export const completeTodo = id => ({ type: types.COMPLETE_TODO, id })
export const completeAllTodos = () => ({ type: types.COMPLETE_ALL_TODOS })
export const clearCompleted = () => ({ type: types.CLEAR_COMPLETED })
export const setVisibilityFilter = filter => ({ type: types.SET_VISIBILITY_FILTER, filter})
// export const getDBTodo = () => ( {type: types.LOAD_DB_TODO,  payload: "abc"})
// export const getDBTodo = () => ({getDBTodo1})
export const getDBTodo = () => ( {type: types.LOAD_DB_TODO,  payload: fetchTodos()})