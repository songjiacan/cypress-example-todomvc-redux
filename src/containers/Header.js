import { connect } from 'react-redux'
import Header from '../components/Header'
import { addTodo  } from '../actions'
import { fetchTodos,addTodoToDB } from '../actions/GetTodo'
import { getMaxId } from '../selectors'

const mapStateToProps = state => ({
    maxId: getMaxId(state)
  })

export default connect(mapStateToProps, { addTodo, fetchTodos, addTodoToDB })(Header)
