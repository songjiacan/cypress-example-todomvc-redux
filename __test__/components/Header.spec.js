import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Header from '../../src/components/Header'
import TodoTextInput from '../../src/components/TodoTextInput'
import { addTodo } from '../../src/actions'

configure({adapter: new Adapter()});

const setup = () => {
  const props = {
    addTodo: jest.fn()
  }
  
  const renderer = createRenderer();
  renderer.render(<Header {...props} />)
  const output = renderer.getRenderOutput()

  return {
    props: props,
    output: output,
    renderer: renderer
  }
}

describe('components', () => {
  describe('Header', () => {

    it('matches snapshot', () => {
      const {output, props } = setup()
      const tree = renderer
        .create(<Header {...props} />)
        .toJSON();
      expect(tree).toMatchSnapshot();

    });

    it('should render correctly', () => {
      const { output } = setup()
      expect(output.type).toBe('header')
      expect(output.props.className).toBe('header')

      const [ h1, input ] = output.props.children
      expect(h1.type).toBe('h1')
      expect(h1.props.children).toBe('todos')
      expect(input.type).toBe(TodoTextInput)
      expect(input.props.newTodo).toBe(true)
      expect(input.props.placeholder).toBe('What needs to be done?')
    })

    it('should call addTodo if length of text is greater than 0', () => {
      const { output, props } = setup()
      const input = output.props.children[1]
      input.props.onSave('')
      expect(props.addTodo).not.toBeCalled()
      input.props.onSave('Use Redux')
      expect(props.addTodo).toBeCalled()
    })

    test('match snapshot with enzyme', () => {
      const {output, props } = setup()
      
      let component = shallow(<Header {...props}  />);
    
      // expect(header.text()).toEqual('Off');
      // cy.get('header input').type('Use Redux{enter}')
      // cy.get('@addTodo').should('have.been.called')
      expect(component).toMatchSnapshot();
      // expect(component.find('input').placeholder()).toEqual('');
      // component
      //     .find('input').get(0)
      //     .simulate('keydown', { which: 'abc' });
      //     // .simulate('click');    
      // expect(props.addTodo).toHaveBeenCalled();
      
    });

  })
})
