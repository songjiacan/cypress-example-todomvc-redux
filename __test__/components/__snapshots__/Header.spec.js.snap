// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header match snapshot with enzyme 1`] = `ShallowWrapper {}`;

exports[`components Header matches snapshot 1`] = `
<header
  className="header"
>
  <h1>
    todos
  </h1>
  <input
    autoFocus={true}
    className="new-todo"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
</header>
`;
